test/client_test.rb

```diff
@@ -1,42 +1,136 @@
 require File.expand_path('../test_helper', __FILE__)
 
-class ClientTest < MiniTest::Unit::TestCase
-  def setup
-    super
-
-    @test_ref = Fauna::Ref.new("classes/#{RandomHelper.random_string}/#{RandomHelper.random_number}")
-    @test_set_match = RandomHelper.random_string
-    @test_set_index = Fauna::Ref.new("indexes/#{RandomHelper.random_string}")
-    @test_obj_key = RandomHelper.random_string
-    @test_obj_value = RandomHelper.random_string
-
-    @stubs.get('tests/ref') do
-      [200, @test_headers, { 'resource' => @test_ref.to_hash }.to_json]
-    end
-    @stubs.get('tests/set') do
-      [200, @test_headers, { 'resource' => Fauna::Set.new('match' => @test_set_match, 'index' => @test_set_index).to_hash }.to_json]
-    end
-    @stubs.get('tests/obj') do
-      [200, @test_headers, { 'resource' => { '@obj' => { @test_obj_key => @test_obj_value } } }.to_json]
-    end
-  end
-
+class ClientTest < FaunaTest # rubocop:disable Metrics/ClassLength
   def test_decode_ref
-    response = @test_client.get('tests/ref')
+    test_ref = Fauna::Ref.new("classes/#{RandomHelper.random_string}/#{RandomHelper.random_number}")
+
+    test_client = stub_client :get, 'tests/ref', [200, {}, { resource: test_ref.to_hash }.to_json]
+    response = test_client.get('tests/ref')
     assert response.is_a?(Fauna::Ref)
-    assert_equal @test_ref.value, response.value
+    assert_equal test_ref, response
   end
 
-  def test_decode_set
-    response = @test_client.get('tests/set')
+  def test_decode_set # rubocop:disable Metrics/AbcSize
```

Comment:
It's a metric set that counts assignments, branches, and conditionals, and complains if there's too many.
Tbh I wouldn't mind disabling that metric everywhere.